hack/cmd/bumpso/bumpso.go

```diff
@@ -0,0 +1,174 @@
+package main
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/coreos/go-semver/semver"
+	"gopkg.in/yaml.v3"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+func main() {
+	if err := run(); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func run() error {
+	project := make(map[string]interface{}, 8)
+	wd, err := os.Getwd()
+	if err != nil {
+		log.Fatal(err)
+	}
+	projectPath := filepath.Join(wd, "olm-catalog/serverless-operator/project.yaml")
+
+	flag.StringVar(&projectPath, "project-path", projectPath, "")
+	flag.Parse()
+
+	var node yaml.Node
+
+	file, err := os.ReadFile(projectPath)
+	if err != nil {
+		return fmt.Errorf("failed to read file %s: %w", projectPath, err)
+	}
+
+	if err := yaml.NewDecoder(bytes.NewBuffer(file)).Decode(&project); err != nil {
+		return fmt.Errorf("failed to decode file into map: %w", err)
+	}
+	if err := yaml.NewDecoder(bytes.NewBuffer(file)).Decode(&node); err != nil {
+		return fmt.Errorf("failed to decode file into node: %w", err)
+	}
+
+	previousVersion, err := previousVersion(project)
+	if err != nil {
+		return err
+	}
+
+	currentVersion, err := currentVersion(project)
+	if err != nil {
+		return err
+	}
+	// We don't care about path versions (patch versions are potentially skipped, etc)
+	currentVersion.Patch = 0
+
+	newVersion := &semver.Version{
+		Major:      currentVersion.Major,
+		Minor:      currentVersion.Minor,
+		Patch:      currentVersion.Patch,
+		PreRelease: currentVersion.PreRelease,
+		Metadata:   currentVersion.Metadata,
+	}
+	newVersion.BumpMinor()
```

Comment:
I wonder if should detect something like `CURRENT_BRANCH==release-*` and bump patch version. But it's kind of niche use case with usual cadence. :)

Reply:
Yes, we can iterate, I felt like spending time on patch releases support was not worth at the initial iteration